### PR TITLE
users: create the home directory when creating a new user

### PR DESCRIPTION
**Pull request by kalbasit**

### CODE REVIEW

File: modules/users/default.nix [hunk 1]
```diff
@@ -137,6 +137,8 @@ in
           dscl . -create '/Users/${v.name}' RealName '${v.description}'
           dscl . -create '/Users/${v.name}' NFSHomeDirectory '${v.home}'
           dscl . -create '/Users/${v.name}' UserShell '${v.shell}'
+          mkdir -p '${v.home}'
+          chown '${toString v.uid}:${toString v.gid}' '${v.home}'
```
Comment (LnL7) [inline modules/users/default.nix]: `/var/empty` is the home directory of multiple nixbld users and is not supposed to be writable in that case.

Comment (kalbasit) [reply modules/users/default.nix]: oh, that's right. I'll chown it conditionally on not `/var/empty`.

Comment (kalbasit) [reply modules/users/default.nix]: I've updated the PR and will mark it as ready for review!

Comment (LnL7) [reply modules/users/default.nix]: I would rather add an option for it instead, also take a look at `man 1 createhomedir`.

Comment (kalbasit) [reply modules/users/default.nix]: Done, PTAL.

**Pull request merged**